src-clojure/me/tonsky/persistent_sorted_set/protocol.cljs

```diff
@@ -0,0 +1,7 @@
+(ns me.tonsky.persistent-sorted-set.protocol)
+
+(defprotocol IStorage
+  (restore [this address])
+  (accessed [this address])
+  (store [this node address])
```

Comment:
I’m not sure what `address` here does? Clojure version only takes node and storage chooses and returns address, so that it can be compatible with e.g. auto-increment. I think we should keep it consistent between versions

Reply:
The main idea here is to reuse the existing address to reduce the storage usage and get rid of gc if possible.
We observed 0.1MB increases for db.sqlite when editing one block.

I agree that we should keep it consistent between versions, maybe someone can implement the same idea with the Clojure version, or I can remove both `address` and `_dirty`.

Reply:
Wait, you can’t reuse addresses though? Because user can keep a reference to an old version of database that is referencing the old version of the block? If you rewrite it the old db ref will stop working?

There’s actually a whole deal about it in Java version, you kind of need to know all alive references to run GC and not break any of them. That was the initial promise of Datomic and Datascript — dbs are immutable, if you keep a reference to them they’ll keep working

Reply:
I'm sorry for not getting back to you sooner.
What I did is to keep using any existing address instead of generating a new address for each branch node and leaf, Clojure version of Datascript has to remove all the unused addresses [here](https://github.com/tonsky/datascript/blob/master/src/datascript/storage.clj#L208), those unused addresses increases the storage usage unless they're garbage collected.